lib/ripper-tags.rb

```diff
@@ -122,6 +124,9 @@ def self.process_args(argv, run = method(:run))
       elsif !options.recursive then abort(optparse.banner)
       end
       options.format ||= File.basename(options.tag_file_name) == "TAGS" ? "emacs" : "vim"
+      if options.format == 'emacs' and not options.provide_tag_file_name
```

Comment:
This solution looks good and tests look good. However, for sake of simplicity, instead of adding new boolean `provide_tag_file_name`, could we simply have `tag_file_name` be `nil` in defaults? Then the logic after parsing would be (pseudo-code):

```
IF tag_file_name IS NIL
  tag_file_name = IF EMACS THEN "TAGS" ELSE "./tags"
END
```